classification/dataset_miniimagenet.py

```diff
@@ -160,19 +162,26 @@ def __getitem__(self, index):
         # pre-process the images and save as numpy arrays (makes the code run much faster afterwards)
         for i, filename in enumerate(filenames_support_x):
             filename_preprocesses = filename[:-4] + '_preprocesses_{}'.format(self.imsize)
-            path_preprocesses = os.path.join(self.path_preprocessed, filename_preprocesses)
+            path_preprocesses = os.path.join(self.path_preprocessed,  filename[:9], filename_preprocesses)
```

Comment:
Why are you creating one subfolder per image? If we split the preprocessed files also in train/test/val subfolders then it should be ok right? So is it ok if I remove this part?

Reply:
i created image_processed, inside it is train, val, test, inside them are folders for classes, inside them are images.
but i think the better way is to do this with pickle, i found three pickle files for miniimagenet train, test, val, it would make life more easy, and preprocessing would not be required.

Reply:
Ok I see, thanks. I'll push an updated version of the code in a bit (I'll keep the old and new option of providing mini-imagenet data so people can choose what works better for them).

The reason I'm saving the preprocessed images is that reading the transformed images from disk is faster than transforming them on-the-fly, and thus speeds up training. 

Cross-domain image classification is really interesting (see also the meta-dataset paper), but I haven't tested CAVIA on this.